src/main/java/uk/gov/companieshouse/controller/PaymentController.java

```diff
@@ -84,7 +84,7 @@ public void patchPaymentData(
 
         if (PaymentStatus.PAID.equals(body.getStatus())) {
             try {
-                dissolutionService.updatePaymentStatus(body, companyNumber);
+                dissolutionService.updatePaymentAndSubmissionStatus(body, companyNumber);
```

Comment:
we could call this `handlePayment` to keep it flexible enough to extend in future if we ever need to add more logic i.e. emailing.